Sources/Web3Core/KeystoreManager/BIP32HDNode.swift

```diff
@@ -22,12 +22,21 @@ extension UInt32 {
 }
 
 public class HDNode {
+    static var maxIterationIndex = UInt32(1) << 31
+
     public struct HDversion {
         // swiftlint:disable force_unwrapping
-        public var privatePrefix: Data = Data.fromHex("0x0488ADE4")!
-        public var publicPrefix: Data = Data.fromHex("0x0488B21E")!
+        public var privatePrefix: Data = Data.fromHex("0x0488ADE4") ?? Data()
+        public var publicPrefix: Data = Data.fromHex("0x0488B21E") ?? Data()
```

Comment:
The `force_unwrapping` rule is disabled for these two lines, so what is the point of empty `Data`?
And if for some reason `fromHex` fails and we get empty `Data` instead the following uses will do the wrong thing: 

<img width="471" alt="Screenshot 2023-03-09 at 11 18 46" src="https://user-images.githubusercontent.com/36865532/223976596-497c1bc6-7b50-4a57-b41f-f397a6e78b1f.png">

<img width="562" alt="Screenshot 2023-03-09 at 11 19 42" src="https://user-images.githubusercontent.com/36865532/223976849-eba72835-1387-4e2f-953e-f900570f320f.png">

Reply:
we should not be doing force unwrap, it could crash. would you prefer me to mark it as optional?

Reply:
I personally would prefer it to crash.
There are situations where crashing is better because silencing the issue won't help you find the bug.
Additionally to that, we have quite a few tests that also use `Data.fromHex(...)`:

<img width="348" alt="Screenshot 2023-03-09 at 20 18 21" src="https://user-images.githubusercontent.com/36865532/224118975-e45cb587-d1be-4c01-98cf-af80e8dad068.png">

These must catch the issue earlier that we will release the version with a bug.

I understand your point of view, but I do not think we will release a version where `Data.fromHex("0x0488ADE4")!` will crash unless we are not really doing PR reviews. 

I'll take a little more time to think about what would be the best way to go.

Reply:
I personally would prefer it to crash.
There are situations where crashing is better because silencing the issue won't help you find the bug.
Additionally to that, we have quite a few tests that also use `Data.fromHex(...)`:

<img width="349" alt="Screenshot 2023-03-09 at 20 25 05" src="https://user-images.githubusercontent.com/36865532/224120381-f54873ff-293c-4d24-be9c-4c3f758f8f15.png">

These must catch the issue earlier that we will release the version with a bug.

I understand your point of view, but I do not think we will release a version where `Data.fromHex("0x0488ADE4")!` will crash unless we are not really doing PR reviews.

I'll take a little more time to think about what would be the best way to go.

Reply:
I see your point that a define unchanging value should produce the same result (a non nil value) every time it is executed